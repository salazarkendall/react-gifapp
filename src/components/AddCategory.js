import React, { useState } from 'react';
import PropTypes from 'prop-types';

/**
 * Este es el Functional Component encargado del funcionamiento del input
 */
export const AddCategory = ({ setCategories }) => {
	/**
	 * Lo siguiente es un hook comun, encargado de la funcionalidad del input
	 */
	const [inputValue, setInputValue] = useState('');

	const handleInputChange = (e) => {
		setInputValue(e.target.value);
	};
	const handleSubmit = (e) => {
		e.preventDefault();

		if (inputValue.trim().length > 2) {
			setCategories((categories) => [...categories, inputValue]);
			setInputValue('');
		}
	};

	return (
		<form onSubmit={handleSubmit}>
			<input
				type="text"
				value={inputValue}
				onChange={handleInputChange}
			/>
		</form>
	);
};

AddCategory.propTypes = {
	setCategories: PropTypes.func.isRequired,
};
