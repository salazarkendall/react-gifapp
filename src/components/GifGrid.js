import React from 'react';

export const GifGrid = ({ category }) => {
	const getGifs = async () => {
		const url = `https://api.giphy.com/v1/gifs/search?q=Dark&limit=10&api_key=GGTJKrckkPhdSGdOc5ZFbTgpFCrcuuNp`;
		const respuesta = await fetch(url);
		const { data } = await respuesta.json();

		const gifs = data.map((imagen) => {
			return {
				id: imagen.id,
				title: imagen.title,
				url: imagen.images?.downsized_medium.url,
			};
		});

		console.log(gifs);
	};

	getGifs();

	return (
		<>
			<h3>{category}</h3>
		</>
	);
};
