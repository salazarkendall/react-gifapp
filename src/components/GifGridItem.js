import React from 'react';
import PropTypes from 'prop-types';
/**
 * Se encarga de darle la estructura HTML a cada uno de los gifs que se mostraran en pantalla
 * Los unicos valores recibidor por parametro son title y url porque asi fue especificado al usar el operador spread
 */
export const GifGridItem = ({ title, url }) => {
    return (
        <div className="card animate__animated animate__zoomIn">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    );
};

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
};
