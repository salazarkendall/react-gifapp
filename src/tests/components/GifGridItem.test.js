import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

/**
 * El componente que vamos a probar tiene props que debemos cumplir
 * Por esto, dichos props se definen una unica vez, fuera de todas las pruebas
 */

describe('Pruebas en GifGridItem', () => {
	let title = 'Este es solo un titulo de ejemplo';
	let url = 'url';
	let wrapper = shallow(<GifGridItem title={title} url={url} />);

	test('Se debe de mostrar el componente correctamente', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('El componente debe de tener un parrafo con el title', () => {
		const p = wrapper.find('p');
		expect(p.text()).toEqual(title);
	});

	test('La imagen debe de tener el url y el alt segun los props', () => {
		const img = wrapper.find('img');
		expect(img.prop('src')).toEqual(url);
		expect(img.prop('alt')).toEqual(title);
	});

	test('La clase del div debe de contener una animacion', () => {
		/**
		 * Estas son 2 formas de hacer lo mismo, pero pueden tener aplicaciones diferentes en el futuro
		 * Lo que realizamos es buscar si un elemento de nuestro componente contiene la clase especificada
		 */
		const div = wrapper.find('div');
		const className = div.prop('className');
		expect(className).toContain('animate__zoomIn');
		expect(className.includes('animate__zoomIn')).toBe(true);
	});
});
