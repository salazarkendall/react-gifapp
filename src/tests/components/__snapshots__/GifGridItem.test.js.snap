// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem Se debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__zoomIn"
>
  <img
    alt="Este es solo un titulo de ejemplo"
    src="url"
  />
  <p>
    Este es solo un titulo de ejemplo
  </p>
</div>
`;
